services/offline/src/lib/online-status.ts

```diff
@@ -24,25 +24,26 @@ interface OnlineStatus {
  * @param {Number} [options.debounceDelay] - Timeout delay to debounce updates, in ms
  * @returns {Object} `{ online, offline }` booleans. Each is the opposite of the other.
  */
-export function useOnlineStatus(options?: OnlineStatusOptions): OnlineStatus {
+export function useOnlineStatus(
+    options: OnlineStatusOptions = {}
+): OnlineStatus {
     // initialize state to `navigator.onLine` value
     const [online, setOnline] = useState(navigator.onLine)
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
     const updateState = useCallback(
         debounce(
             ({ type }: Event) => setOnline(type === 'online'),
-            options?.debounceDelay || 1000
+            options.debounceDelay || 1000
         ),
-        [options]
+        [options.debounceDelay]
     )
 
     // on 'online' or 'offline' events, set state
     useEffect(() => {
         window.addEventListener('online', updateState)
         window.addEventListener('offline', updateState)
         return () => {
-            updateState.cancel()
```

Comment:
Removing this will cause problems on unmount, because the debounced callback will fire after the component is unmounted which is not allowed.  As such I think this cancellation is still needed... it should only run when `debounceDelay` is changed or when the component is unmounted

Reply:
Maybe this needs to call `updateState.flush()` instead so that any pending update is applied?

Reply:
https://docs-lodash.com/v4/debounce/

Reply:
I also think flush is the more correct option

Reply:
Changed to `updateState.flush()` 👍